### ISSUE TITLE
led blinks but Not getting any signal in all the three dedicated pins. 

### ISSUE THREAD

**Issue by AbidHussain70** [spenceraxani/CosmicWatch-Desktop-Muon-Detector-v2#56]
Hi, first wanna say thanks for such a beautiful and well documented project. 
I follow all of your steps and exactly copy your work. I did all the troubleshooting, DC-DC booster working fine, OLED working, SD card, and led also works. The problem is, that when I attached the SiPM PCB + plastic scintillator on main board the led does start blinking, but I am not getting any pulse on BNC or any of the three pins dedicated for checking pulses. I am confused, on one side no pulse while on the other side the led blinks. 
Can you help please.

Another thing I just missed one test, I did not check the plastic scintillator with torch. Which torch should we use for this purpose.
Looking for your answer thinks in advance.

**Comment by spenceraxani** [comment]
Hello,

If you have measured the DC-DC booster output and are seeing 29.6V, but are not seeing any pulses on through the BNC connector, then it is an issue with the SiPM PCB. The most likely two candidates are that either the SiPM was mounted backwards, or perhaps the connection on one of the SiPM pins came loose while securing it to the scintillator. 

Seeing blinking seems odd though. If it is rhythmic, then perhaps you have the wrong code uploaded? Although that wouldn’t explain why you don’t see pulses from the BNC connection. It could be triggering on noise or spurious signals from the SiPM PCB, which could look random and not like SiPM pulses, this could be explained by a broken connection on the SIPM pin. Sadly, I think you’ll have to unwrap your scintillator to check it.

Thanks,

Spencer

**Comment by AbidHussain70** [comment]
Thank you for quick response.
 """"........ If you have measured the DC-DC booster output and are seeing 29.6V, but are not seeing any pulses on through the BNC connector, then it is an issue with the SiPM PCB. The most likely two candidates are that either the SiPM was mounted backwards, or perhaps the connection on one of the SiPM pins came loose while securing it to the scintillator......."""

Yes DC-DC booster output is giving 29.65V.
![image](https://user-images.githubusercontent.com/23444211/76274860-bba8d880-62bc-11ea-868d-cd1af8070501.png)
For the second part of your comment, I checked it several time. I solder 2 SiPM PCBs, but same problem. 
Check the picture pin 1 of the SiPM has a small extra pin as you said in video 3 (https://youtu.be/yFgin5wlw4I).
![image](https://user-images.githubusercontent.com/23444211/76275164-8fda2280-62bd-11ea-9be5-255921f90647.png)

And in the SiPM PCB pin one is labeled with small circle.
![image](https://user-images.githubusercontent.com/23444211/76275425-3d4d3600-62be-11ea-95b7-8d72d8384148.png)

I solder the SiPM exactly in the same direction. And I also checked the resistance, in one direction it is giving some finite value (in Mega ohms) in other direction no value. Please check the pictures.

![image](https://user-images.githubusercontent.com/23444211/76275589-9fa63680-62be-11ea-857d-54b2b11dc859.png)

![image](https://user-images.githubusercontent.com/23444211/76275632-b3519d00-62be-11ea-96fd-144ce3a5d734.png)

 
And from datasheet of the SiPM pin 1 is anode and pin 3 is cathode. 

![image](https://user-images.githubusercontent.com/23444211/76275072-4b4e8700-62bd-11ea-8828-020b36dbaa0e.png)

I check the voltage drop on anode and cathode pin of the SiPM. Which was working.

![image](https://user-images.githubusercontent.com/23444211/76275777-193e2480-62bf-11ea-8632-12a9c0f56b96.png)

 
 """"........ Seeing blinking seems odd though. If it is rhythmic, then perhaps you have the wrong code uploaded? Although that wouldn’t explain why you don’t see pulses from the BNC connection. It could be triggering on noise or spurious signals from the SiPM PCB, which could look random and not like SiPM pulses, this could be explained by a broken connection on the SIPM pin. Sadly, I think you’ll have to unwrap your scintillator to check it..........."""""

 Blinking in rhythm, well I can not say anything about this, but it does blink, sometimes first and sometimes slow.  I upload OLED code from v2. I unwrap it, in fact I did unwrap it several time.

**Comment by spenceraxani** [comment]
Hi,

It actually sounds like your detector is working. Could it be that the issue is with the oscilloscope? Perhaps the horizontal center if off-scope, or something like this? Just to ensure you are looking for the correct signal, it should be a positive pulse, 10s of mV in voltage, and order 500ns in width. When you use the OLED screen, what is the approximate count rate that you see?

I forgot to reply to your question about the scintillator. I use a hot air gun from a soldering station to melt the scintillator. This isn’t really needed. If you have a good optical surface between the SiPM and scintillator, improving the reflection on the sides won’t dramatically help collect photons.

Thanks,

Spencer

**Comment by AbidHussain70** [comment]
Sorry I forgot to close the issue. The problem was with SiPM soldering. Though I did troubleshooting as you mentioned in your video and in the document, the SiPM circuit was showing same results. But due to the SiPM tiny pins, it became disconnected somehow during installation of SiPM circuit on main board.  
I used Rosin Flux to fix the problem. Now the SiPM is solder properly and circuit is working fine.
Thank you.
![image](https://user-images.githubusercontent.com/23444211/79179725-dbdb3280-7e3a-11ea-9ecf-efa54465a5fe.png)

![image](https://user-images.githubusercontent.com/23444211/79179755-ebf31200-7e3a-11ea-801c-7599d3d86d6e.png)

.
